[bug] "cargo install tauri-cli --version "^2.0.0-beta"" fails to build on Windows 11

### Describe the bug

As stated in the title the command ` cargo install tauri-cli --version "^2.0.0-beta"` fails to build on Windows 11.

### Reproduction

Run `cargo install tauri-cli --version "^2.0.0-beta"` attempting to install beta.14

### Expected behavior

It installs without issue

### Full `tauri info` output

```text
[✔] Environment
    - OS: Windows 10.0.22635 X64
    ✔ WebView2: 124.0.2478.67
    ✔ MSVC: Visual Studio Community 2022
    ✔ rustc: 1.79.0-nightly (ef8b9dcf2 2024-04-24)
    ✔ cargo: 1.79.0-nightly (c93926759 2024-04-23)
    ✔ rustup: 1.27.0 (bbb9276d2 2024-03-08)
    ✔ Rust toolchain: nightly-x86_64-pc-windows-msvc (environment override by RUSTUP_TOOLCHAIN)
    - node: 20.8.0
    - pnpm: 8.11.0
    - yarn: 1.22.19
    - npm: 10.5.1

[-] Packages
    - tauri [RUST]: 2.0.0-beta.16
    - tauri-build [RUST]: 2.0.0-beta.13
    - wry [RUST]: 0.39.3
    - tao [RUST]: 0.27.1
    - tauri-cli [RUST]: 2.0.0-beta.0
    - @tauri-apps/api [NPM]: 2.0.0-beta.9
    - @tauri-apps/cli [NPM]: 2.0.0-beta.0

[-] App
    - build-type: bundle
    - CSP: unset
    - frontendDist: ../.output/public
    - devUrl: http://localhost:3000/
    - framework: Vue.js (Nuxt)
    - bundler: Vite
```
```


### Stack trace

```text
error[E0308]: mismatched types
   --> C:\Users\{{user}}\.cargo\registry\src\index.crates.io-6f17d22bba15001f\tauri-cli-2.0.0-beta.14\src\helpers\flock.rs:343:28
    |
343 |       let ret = UnlockFile(file.as_raw_handle(), 0, 0, !0, !0);
    |                 ---------- ^^^^^^^^^^^^^^^^^^^^ expected `winapi::ctypes::c_void`, found `std::ffi::c_void`
    |                 |
    |                 arguments to this function are incorrect
    |
    = note: `std::ffi::c_void` and `winapi::ctypes::c_void` have similar names, but are actually distinct types
note: `std::ffi::c_void` is defined in crate `core`
   --> C:\Users\{{user}}\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\lib/rustlib/src/rust\library\core\src\ffi\mod.rs:183:1
    |
183 | pub enum c_void {
    | ^^^^^^^^^^^^^^^
note: `winapi::ctypes::c_void` is defined in crate `winapi`
   --> C:\Users\{{user}}\.cargo\registry\src\index.crates.io-6f17d22bba15001f\winapi-0.3.9\src\lib.rs:38:5
    |
38  |     pub enum c_void {}
    |     ^^^^^^^^^^^^^^^
note: function defined here
   --> C:\Users\{{user}}\.cargo\registry\src\index.crates.io-6f17d22bba15001f\winapi-0.3.9\src\um\fileapi.rs:519:12
    |
519 |     pub fn UnlockFile(
    |            ^^^^^^^^^^

error[E0308]: mismatched types
   --> C:\Users\{{user}}\.cargo\registry\src\index.crates.io-6f17d22bba15001f\tauri-cli-2.0.0-beta.14\src\helpers\flock.rs:355:28
    |
355 |       let ret = LockFileEx(file.as_raw_handle(), flags, 0, !0, !0, &mut overlapp...
    |                 ---------- ^^^^^^^^^^^^^^^^^^^^ expected `winapi::ctypes::c_void`, found `std::ffi::c_void`
    |                 |
    |                 arguments to this function are incorrect
    |
    = note: `std::ffi::c_void` and `winapi::ctypes::c_void` have similar names, but are actually distinct types
note: `std::ffi::c_void` is defined in crate `core`
   --> C:\Users\{{user}}\.rustup\toolchains\nightly-x86_64-pc-windows-msvc\lib/rustlib/src/rust\library\core\src\ffi\mod.rs:183:1
    |
183 | pub enum c_void {
    | ^^^^^^^^^^^^^^^
note: `winapi::ctypes::c_void` is defined in crate `winapi`
   --> C:\Users\{{user}}\.cargo\registry\src\index.crates.io-6f17d22bba15001f\winapi-0.3.9\src\lib.rs:38:5
    |
38  |     pub enum c_void {}
    |     ^^^^^^^^^^^^^^^
note: function defined here
   --> C:\Users\{{user}}\.cargo\registry\src\index.crates.io-6f17d22bba15001f\winapi-0.3.9\src\um\fileapi.rs:440:12
    |
440 |     pub fn LockFileEx(
    |
```
```


### Additional context

I am trying to upgrade Tauri CLI

--- a/.changes/cli-windows-sys.md
+++ b/.changes/cli-windows-sys.md
@@ -1,0 +1,5 @@
+---
+"tauri-cli": "patch"
+---
+
+Use `windows-sys` crate instead of `winapi` which fixes installing the published cli from crates.io using `cargo install tauri-cli --version "^2.0.0-beta"`.

--- a/.changes/cli-windows-sys.md
+++ b/.changes/cli-windows-sys.md
@@ -1,5 +1,5 @@
 ---
-"tauri-cli": "patch"
+"tauri-cli": "patch:bug"
 ---
 
 Use `windows-sys` crate instead of `winapi` which fixes installing the published cli from crates.io using `cargo install tauri-cli --version "^2.0.0-beta"`.

--- a/tooling/cli/Cargo.toml
+++ b/tooling/cli/Cargo.toml
@@ -96,9 +96,13 @@ resvg = "0.40.0"
 dunce = "1"
 glob = "0.3"
 
-[target."cfg(windows)".dependencies]
-winapi = { version = "0.3", features = [ "handleapi", "processenv", "winbase", "wincon", "winnt" ] }
-cc = "1"
+[target."cfg(windows)".dependencies.windows-sys]
+version = "0.52"
+features = [
+  "Win32_Storage_FileSystem",
+  "Win32_System_IO",
+  "Win32_System_Console"
+]
 
 [target."cfg(unix)".dependencies]
 libc = "0.2"

--- a/tooling/cli/src/helpers/flock.rs
+++ b/tooling/cli/src/helpers/flock.rs
@@ -305,10 +305,10 @@ mod sys {
   use std::mem;
   use std::os::windows::io::AsRawHandle;
 
-  use winapi::shared::minwindef::DWORD;
-  use winapi::shared::winerror::{ERROR_INVALID_FUNCTION, ERROR_LOCK_VIOLATION};
-  use winapi::um::fileapi::{LockFileEx, UnlockFile};
-  use winapi::um::minwinbase::{LOCKFILE_EXCLUSIVE_LOCK, LOCKFILE_FAIL_IMMEDIATELY};
+  use windows_sys::Win32::Foundation::{ERROR_INVALID_FUNCTION, ERROR_LOCK_VIOLATION, HANDLE};
+  use windows_sys::Win32::Storage::FileSystem::{
+    LockFileEx, UnlockFile, LOCKFILE_EXCLUSIVE_LOCK, LOCKFILE_FAIL_IMMEDIATELY, LOCK_FILE_FLAGS,
+  };
 
   pub(super) fn lock_shared(file: &File) -> Result<()> {
     lock_file(file, 0)
@@ -339,25 +339,31 @@ mod sys {
   }
 
   pub(super) fn unlock(file: &File) -> Result<()> {
-    unsafe {
-      let ret = UnlockFile(file.as_raw_handle(), 0, 0, !0, !0);
-      if ret == 0 {
-        Err(Error::last_os_error())
-      } else {
-        Ok(())
-      }
+    let ret = unsafe { UnlockFile(file.as_raw_handle() as HANDLE, 0, 0, !0, !0) };
+    if ret == 0 {
+      Err(Error::last_os_error())
+    } else {
+      Ok(())
     }
   }
 
-  fn lock_file(file: &File, flags: DWORD) -> Result<()> {
-    unsafe {
+  fn lock_file(file: &File, flags: LOCK_FILE_FLAGS) -> Result<()> {
+    let ret = unsafe {
       let mut overlapped = mem::zeroed();
-      let ret = LockFileEx(file.as_raw_handle(), flags, 0, !0, !0, &mut overlapped);
-      if ret == 0 {
-        Err(Error::last_os_error())
-      } else {
-        Ok(())
-      }
+      LockFileEx(
+        file.as_raw_handle() as HANDLE,
+        flags,
+        0,
+        !0,
+        !0,
+        &mut overlapped,
+      )
+    };
+
+    if ret == 0 {
+      Err(Error::last_os_error())
+    } else {
+      Ok(())
     }
   }
 }

--- a/tooling/cli/src/interface/rust/desktop.rs
+++ b/tooling/cli/src/interface/rust/desktop.rs
@@ -446,12 +446,14 @@ mod terminal {
 #[cfg(windows)]
 mod terminal {
   use std::{cmp, mem, ptr};
-  use winapi::um::fileapi::*;
-  use winapi::um::handleapi::*;
-  use winapi::um::processenv::*;
-  use winapi::um::winbase::*;
-  use winapi::um::wincon::*;
-  use winapi::um::winnt::*;
+
+  use windows_sys::Win32::{
+    Foundation::{CloseHandle, GENERIC_READ, GENERIC_WRITE, INVALID_HANDLE_VALUE},
+    Storage::FileSystem::{CreateFileA, FILE_SHARE_READ, FILE_SHARE_WRITE, OPEN_EXISTING},
+    System::Console::{
+      GetConsoleScreenBufferInfo, GetStdHandle, CONSOLE_SCREEN_BUFFER_INFO, STD_ERROR_HANDLE,
+    },
+  };
 
   pub fn stderr_width() -> Option<usize> {
     unsafe {
@@ -465,13 +467,13 @@ mod terminal {
       // INVALID_HANDLE_VALUE. Use an alternate method which works
       // in that case as well.
       let h = CreateFileA(
-        "CONOUT$\0".as_ptr() as *const CHAR,
+        "CONOUT$\0".as_ptr(),
         GENERIC_READ | GENERIC_WRITE,
         FILE_SHARE_READ | FILE_SHARE_WRITE,
         ptr::null_mut(),
         OPEN_EXISTING,
         0,
-        ptr::null_mut(),
+        0,
       );
       if h == INVALID_HANDLE_VALUE {
         return None;

